cli/ast.rs

```diff
@@ -135,8 +135,8 @@ impl DiagnosticBuffer {
 #[derive(Debug, Clone)]
 pub struct ErrorBuffer(Arc<RwLock<Vec<Diagnostic>>>);
 
-impl ErrorBuffer {
-  pub fn new() -> Self {
```

Comment:
aside: shouldn't this result in a clippy error currently?

Reply:
I would expect so, but it seems that's only the case when having `lib.rs`